image/image.go

```diff
@@ -122,6 +148,31 @@ type genericManifest interface {
 	GetLayers() []string
 }
 
+type descriptor struct {
+	MediaType string `json:"mediaType"`
+	Size      int64  `json:"size"`
+	Digest    string `json:"digest"`
+}
+
+type manifestSchema2 struct {
+	Config descriptor   `json:"config"`
+	Layers []descriptor `json:"layers"`
+}
+
+func (m *manifestSchema2) String() string {
+	// TODO: does it make sense to have string here?
```

Comment:
Sure, why not. I don’t know the history behind `skopeo layers` and I didn’t take the time to understand it, so all my refactoring was aiming at isolating the parts I don’t understand without changing their behavior; anything you decide to do with that command is fine with me.